pkg/exdns/exdns.go

```diff
@@ -39,3 +40,19 @@ func Decanonize(qName string) string {
 	}
 	return qName
 }
+
+// IsDescendent checks if one fully qualified name is
+// a descendent of another.
+func IsDescendent(ancestor, child string) bool {
+	s, ok := strings.CutSuffix(child, ancestor)
+	switch {
+	case !ok:
+		return false
+	case ancestor == ".":
+		return true
+	case s == "", s[len(s)-1] == '.':
+		return true
+	default:
+		return false
+	}
+}
```

Comment:
isn't this an alternative and a bit more fragile implementation of [dns.IsSubdomain(parent, child string) bool )](https://pkg.go.dev/github.com/miekg/dns#IsSubDomain) ?

If yes (and as far as I can see it is.) please remove it.

Reply:
I had the impression `IsSubdomain()` only accepts direct descendants, am I wrong?

Reply:
I am not sure what you mean by "direct descendants" but `IsSubdomain()` will check if `ancestor` can be removed from `child`. So will return true for `(example.com.,test1.example.com)` and `(example.com, test2.test1.example.com)` and `(test1.example.com, test2.test1.example.com)` but false for `(example.net, example.com)` or `(example.net, test1.anotherexample.net)`